SpatialGDK/Source/SpatialGDK/Public/SpatialCommonTypes.h

```diff
@@ -64,3 +65,30 @@ struct TWeakObjectPtrKeyFuncs : DefaultKeyFuncs<ElementType, bInAllowDuplicateKe
 // TODO: These can be removed once event tracing is enabled UNR-3981
 using FWorkerComponentUpdate = FTrackableWorkerType<Worker_ComponentUpdate>;
 using FWorkerComponentData = FTrackableWorkerType<Worker_ComponentData>;
+
+/* A little helper to ensure no re-entrancy of modifications of data structures. */
+struct FUsageLock
+{
+	bool bIsSet = false;
+
+	struct Scope
+	{
+		bool& bIsSet;
+		Scope(bool& bIsSet)
+			: bIsSet(bIsSet)
+		{
+			ensureMsgf(!bIsSet, TEXT("Unexpected re-entrancy occured in the Spatial GDK."));
```

Comment:
Perhaps a verify so it runs in Shipping?

Reply:
So verify doesn't actually assert in shipping - it only evaluates the expression (I guess if you have an expression with side-effects). I'll add this to the shipping assert ticket